raiden/tests/integration/network/transport/rif_comms/test_client.py

```diff
@@ -90,223 +101,207 @@ def test_connect_peers():
+        for node in nodes:
+            # because node 2 can be already stopped, we cannot call stop() indiscriminately
+            if not node._process.poll():  # if True, node is still running
+                node.stop()
 
 
-@pytest.mark.parametrize("amount_of_nodes", [2])
-def test_subscribe_to_peers(comms_nodes):
-    comms_node_1, comms_node_2 = comms_nodes[1], comms_nodes[2]
-    client_1, address_1 = comms_node_1.client, comms_node_1.address
-    client_2, address_2 = comms_node_2.client, comms_node_2.address
+@pytest.mark.parametrize("nodes_to_clients", [{1: 1}, {1: 2}])
+def test_subscribe_to_invalid(comms_clients):
+    unregistered_address = generate_address()
+    for client in comms_clients.values():
+        # no subscriptions should be present
+        assert client._is_subscribed_to(unregistered_address) is False
+        # subscribe to unregistered address
+        topic_id, _ = client.subscribe_to(unregistered_address)
+        # FIXME: this should probably throw an exception rather than an empty result
+        assert topic_id is ""
+        assert client._is_subscribed_to(unregistered_address) is False
+
+
+@pytest.mark.parametrize("nodes_to_clients", [{1: 1}, {1: 2}, {1: 2, 2: 2}])
+def test_subscribe_to_self(comms_clients):
+    for client in comms_clients.values():
+        # no subscription should be present
+        assert client._is_subscribed_to(client.rsk_address.address) is False
+        # subscribe to self and check subscription
+        topic_id, _ = client.subscribe_to(client.rsk_address.address)
+        assert topic_id
+        assert client._is_subscribed_to(client.rsk_address.address) is True
+
+
+@pytest.mark.xfail(reason="fails, multi addr issue")
+@pytest.mark.parametrize("nodes_to_clients", [{1: 1, 2: 1}, {1: 2}])
+def test_subscribe_to_peers(comms_clients):
+    client_1 = comms_clients[1]
+    client_2 = comms_clients[2]
 
     # no subscriptions should be present
-    assert client_1._is_subscribed_to(address_2) is False
-    assert client_2._is_subscribed_to(address_1) is False
+    assert client_1._is_subscribed_to(client_2.rsk_address.address) is False
+    assert client_2._is_subscribed_to(client_1.rsk_address.address) is False
 
     # subscribe from node 1 to 2 and check subscriptions
-    topic_id_1, _ = client_1.subscribe_to(address_2)
+    topic_id_1, _ = client_1.subscribe_to(client_2.rsk_address.address)
     assert topic_id_1
-    assert client_1._is_subscribed_to(address_1) is False
-    assert client_1._is_subscribed_to(address_2) is True
-    assert client_2._is_subscribed_to(address_1) is False
-    assert client_2._is_subscribed_to(address_2) is False
+    assert client_1._is_subscribed_to(client_1.rsk_address.address) is False
+    assert client_1._is_subscribed_to(client_2.rsk_address.address) is True
+    assert client_2._is_subscribed_to(client_1.rsk_address.address) is False
+    assert client_2._is_subscribed_to(client_2.rsk_address.address) is False # FIXME this fails
```

Comment:
[RLP-977](https://jirainfuy.atlassian.net/browse/RLP-977)

Reply:
will be tested with the PR of update https://github.com/rsksmart/lumino/pull/193